iml-agent/src/action_plugins/stratagem/action_purge.rs

```diff
@@ -63,3 +64,26 @@ pub async fn read_mailbox(
         })
         .await
 }
+
+pub async fn process_fids(
+    (fsname_or_mntpath, _task_args, fid_list): (String, HashMap<String, String>, Vec<FidItem>),
+) -> Result<Vec<FidError>, ImlAgentError> {
+    let llapi = search_rootpath(fsname_or_mntpath).await?;
+
+    let rmfids_size = llapi.rmfids_size();
+
+    let fids = fid_list.into_iter().map(|fi| fi.fid.clone());
+    stream::iter(fids)
+        .chunks(rmfids_size)
+        .map(|xs| Ok::<_, ImlAgentError>(xs.into_iter().collect()))
+        .try_for_each_concurrent(10, |fids| {
+            rm_fids(llapi.clone(), fids)
+                .or_else(|e| {
+                    warn!("Error removing fid {}", e);
+                    future::ok(())
+                })
+                .map_ok(|_| debug!("removed {} fids", rmfids_size))
+        })
+        .await?;
+    Ok(vec![])
```

Comment:
This doesn't return any `FidErrors`, even though `rm_fids` is failable.

Reply:
We don't care.  They can come around again.

Reply:
Then why are we returning something that we know will never be filled? This makes the API confusing at least.

Reply:
Per the standard, all Action calls return Vec<FidError> which gets processed uniformly.